src/lib/support/SparseArray.h

```diff
@@ -0,0 +1,133 @@
+/*
+ *
+ *    Copyright (c) 2020 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+/**
+ *    @file
+ *      Defines a data structure that can store an array of uint64_t values
```

Comment:
So I think @andy31415 is right: this is basically a "set of nonzero uint64", and the fact that it uses an array as a backing store is an implementation detail.  I think it's worth naming accordingly.

Documenting that the base64 serialize/deserialize process can safely cross endianness as part of the API is good. Documenting that the actual storage is LE is probably better done next to that storage, since that's implementation detail.

Reply:
@bzbarsky-apple Any suggestions on the name?

Reply:
Depending on how much we think we will reuse this, maybe `NodeIdSet` and use `NodeId` in the API surface to make it really clear what this is for.

Alternatly, `IntegerSet` for the filename if we expect to have other types of sets in here eventually and `U64Set` for the type name.  Or `U64Set` for both the file and type name?

Reply:
Changed it to `SerializableIntegerSet`